server/src/org/apache/cloudstack/network/topology/AdvancedNetworkVisitor.java

```diff
@@ -61,13 +61,15 @@
     @Override
     public boolean visit(final UserdataPwdRules userdata) throws ResourceUnavailableException {
         final VirtualRouter router = userdata.getRouter();
-
         final Commands commands = new Commands(Command.OnError.Stop);
         final VirtualMachineProfile profile = userdata.getProfile();
         final NicVO nicVo = userdata.getNicVo();
         final UserVmVO userVM = userdata.getUserVM();
 
-        _commandSetupHelper.createPasswordCommand(router, profile, nicVo, commands);
+        if (router.getRedundantState() == VirtualRouter.RedundantState.MASTER) {
```

Comment:
1. line 69 should be changed to 
```
if (!router.getIsRedundantRouter() || router.getRedundantState() == RedundantState.MASTER) {
```

2, this works in a new isolated network ( with RVR), but not working with a new VPC (with RVR). It might because there is always a guest network in new isolated network so VRs are MASTER/BACKUP, but for VPC they are BACKUP/BACKUP. I have no idea how to fix it, maybe we do not change this part and let password stored in both......actually only the first vm in a vpc.